### PR TITLE
CB-18211 Sync DB engine version into Stack and SdxCluster if missing

### PR DESCRIPTION
**Pull request by lacikaaa**
Introduced a new stack patcher for fetching DB engine version where missing.
In case of embedded it runs only if the stack is available as it has to run a salt command to get the actual version.
In case of external it gets the version from redbeams and syncs it regardless the stack status.

The sync runs as a quartz job, using StackPatcher framework in the core. When a Datalake is affected it updates the datala service db
by calling an internal endpoint. This way there is no need to introduce a new quartz job and datalake side.

If anything goes sideways, like the version can't be fetched, then it will be retried another time scheduled by the framework.

See detailed description in the commit message.

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java [hunk 1]
```diff
@@ -1359,4 +1356,12 @@ public FlowIdentifier rotateSaltPassword(SdxCluster sdxCluster) {
         cloudbreakFlowService.saveLastCloudbreakFlowChainId(sdxCluster, flowIdentifier);
         return sdxReactorFlowManager.triggerSaltPasswordRotationTracker(sdxCluster);
     }
+
+    public void updateDatabaseEngineVersion(String crn, String databaseEngineVersion) {
+        int updatedCount = sdxClusterRepository.updateDatabaseEngineVersion(crn, databaseEngineVersion);
+        LOGGER.info("Updated database engine version for [{}] with [{}]", crn, databaseEngineVersion);
```
Comment (Bajzathd) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:1362]: This should be in the _else_ of the following _if_ statement

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/CollectDbEngineVersionPatchService.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    private void updateDbEngineVersion(Stack stack, String dbEngineVersion) {
+        if (stack.isDatalake()) {
+            LOGGER.info("Updating DB engine version in datalake service with crn [{}] to [{}]", stack.getResourceCrn(), dbEngineVersion);
+            sdxClientService.updateDatabaseEngineVersion(stack.getResourceCrn(), dbEngineVersion);
+        }
+        stackService.updateExternalDatabaseEngineVersion(stack.getId(), dbEngineVersion);
+    }
+
+    private Optional<String> collectDbEngineVersion(Stack stack) {
+        if (StringUtils.isBlank(stack.getCluster().getDatabaseServerCrn())) {
```
Comment (Bajzathd) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/CollectDbEngineVersionPatchService.java:74]: nit: it would improve readability if the condition was applied the same way as in the if statement at the beginning of doApply
`if (StringUtils.isNotBlank(stack.getCluster().getDatabaseServerCrn())) {`

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeOrchestratorService.java [hunk 3]
```diff
@@ -88,19 +89,13 @@ public void validateDbDirectorySpace(Long stackId) throws CloudbreakOrchestrator
                 stateParams.getTargetHostNames(), "df -k /dbfs | tail -1 | awk '{print $4}'");
         Map<String, String> dataDirSizeMap = hostOrchestrator.runCommandOnHosts(List.of(stateParams.getPrimaryGatewayConfig()),
                 stateParams.getTargetHostNames(), "du -sk /dbfs/pgsql | awk '{print $1}'");
-        Integer dataDirSize = dataDirSizeMap.entrySet().stream()
```
Comment (Bajzathd) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeOrchestratorService.java:91]: These changes seem unrelated to the rest of PR

Comment (lacikaaa) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeOrchestratorService.java:91]: it's an optimization :) because I had to understand the code, and I did the refactor. the result is exactly the same

**Review by Bajzathd — APPROVED** [review]
